cmd/syncthing/gui.go

```diff
@@ -1567,15 +1567,51 @@ func browseFiles(current string, fsType fs.FilesystemType) []string {
 
 	fs := fs.NewFilesystem(fsType, searchDir)
 
-	subdirectories, _ := fs.Glob(searchFile + "*")
+	subdirectories, _ := fs.DirNames("")
 
-	ret := make([]string, 0, len(subdirectories))
+	type match struct {
+		path             string
+		case_differences int
+	}
+
+	matches := make([]match, 0, len(subdirectories))
+
+next_subdir:
 	for _, subdirectory := range subdirectories {
+		var m match
+		for i := 0; i < len(subdirectory) && i < len(searchFile); i += 1 {
+			if subdirectory[i] == searchFile[i] {
+				continue
+			}
+
+			if strings.EqualFold(string(subdirectory[i]), string(searchFile[i])) {
+				m.case_differences += 1
+				continue
+			}
+
+			continue next_subdir
+		}
+
 		info, err := fs.Stat(subdirectory)
 		if err == nil && info.IsDir() {
-			ret = append(ret, filepath.Join(searchDir, subdirectory)+pathSeparator)
+			m.path = filepath.Join(searchDir, subdirectory) + pathSeparator
+			matches = append(matches, m)
 		}
 	}
+
+	// sort by number of differences with pattern
+	// ties are resolved by alphabetical order
+	sort.Slice(matches, func(i, j int) bool {
+		if matches[i].case_differences == matches[j].case_differences {
+			return strings.ToLower(matches[i].path) < strings.ToLower(matches[j].path)
+		}
+		return matches[i].case_differences < matches[j].case_differences
```

Comment:
This can just be:

return caseDiffB > caseDiffA || lower(a) < lower(b)

Reply:
I believe that would be incorrect. Example: `caseDiffB = 0, caseDiffA = 1` should always return `false` even if `lower(a) < lower(b)`